README.md

```diff
@@ -123,46 +125,177 @@ Additionally, you'll need to copy debug symbols for debugging and crash reportin
 On your application target’s “Build Phases” settings tab, click the “+” icon and choose “New Copy Files Phase”.
 Click the “Destination” drop-down menu and select “Products Directory”. For each framework, drag and drop corresponding dSYM file.
 
-## Usage Examples
 
-Before starting practicing with the usage examples be sure that the SDK is configured. Check out our [SDK configuration guides][_configure_sdk] for more information.
 
-#### Generate key pair using VirgilCrypto
+## Configure SDK
+
+This section contains guides on how to set up Virgil Core SDK modules for authenticating users, managing Virgil Cards and storing private keys.
+
+### Set up authentication
+
+Set up user authentication with tokens that are based on the [JSON Web Token standard](https://jwt.io/) with some Virgil modifications.
+
+In order to make calls to Virgil Services (for example, to publish user's Card on Virgil Cards Service), you need to have a JSON Web Token ("JWT") that contains the user's `identity`, which is a string that uniquely identifies each user in your application.
+
+Credentials that you'll need:
+
+|Parameter|Description|
+|--- |--- |
+|App ID|ID of your Application at [Virgil Dashboard](https://dashboard.virgilsecurity.com)|
+|App Key ID|A unique string value that identifies your account at the Virgil developer portal|
+|App Key|A Private Key that is used to sign API calls to Virgil Services. For security, you will only be shown the App Key when the key is created. Don't forget to save it in a secure location for the next step|
+
+#### Set up JWT provider on Client side
+
+Use these lines of code to specify which JWT generation source you prefer to use in your project:
 
 ```swift
-import VirgilCrypto
+import VirgilSDK
 
-let crypto = try! VirgilCrypto()
+// Get generated token from server-side
+let authenticatedQueryToServerSide: ((String) -> Void) -> Void = { completion in
+    completion("eyJraWQiOiI3MGI0NDdlMzIxZj....MK7p7Ak")
+}
 
-let keyPair = try! crypto.generateKeyPair()
+// Setup AccessTokenProvider
+let accessTokenProvider = CallbackJwtProvider { tokenContext, completion in
+    authenticatedQueryToServerSide { jwtString in
+        completion(jwtString, nil)
+    }
+}
 ```
 
-#### Save and retrieve key using keychain storage
+#### Generate JWT on Server side
```

Comment:
There couldn't be any "server" side in Swift. There are few frameworks, but generally, it's useless part.